containers/src/Kokkos_Vector.hpp

```diff
@@ -21,6 +21,22 @@
 #define KOKKOS_IMPL_PUBLIC_INCLUDE_NOTDEFINED_VECTOR
 #endif
 
+#include <Kokkos_Macros.hpp>
+
+#if defined(KOKKOS_ENABLE_DEPRECATED_CODE_4)
+#if defined(KOKKOS_ENABLE_DEPRECATION_WARNINGS)
+namespace {
+[[deprecated("Deprecated <Kokkos_Vector.hpp> header is included")]] int
```

Comment:
@masterleinad why did you do this rather than `KOKKOS_DEPRECATED_WITH_COMMENT`?

Reply:
What I cared about is using "#error" and including "Kokkos_Macros.hpp" but not so much about using `KOKKOS_DEPRECATED_WITH_COMMENT` as opposed to `KOKKOS_ENABLE_DEPRECATION_WARNINGS`. The way now at least only declares extra symbols when they are needed for the deprecation warning.

Reply:
Good enough.

To be pedantic, there will be no symbol, that is why I used an anonymous namespace.